### DISCUSSION TITLE
Multiple LDAP Domain Authentication

### DISCUSSION THREAD

**Discussion by IdRatherStand** [netbox-community/netbox#10274] [Q&A]
We have a use case where a subset of users that log on via a completely separate domain will need to access Netbox. As far as I can see there are a few ways to achieve this:

1. Adding multiple domains to the LDAP config as per [Django docs](https://django-auth-ldap.readthedocs.io/en/latest/multiconfig.html)
2. Creating local accounts within Netbox for non-primary domain users
3. Creating accounts within the primary domain for these users solely for Netbox authentication purposes

Unfortunately for us 3) is simply not possible due to organisational reasons and 2) is less palatable due to security reasons (password rotation / account lockout etc).

Is anyone aware of whether 1) is still possible within Netbox? I've taken a look and seem to remember Netbox directly using the Django LDAP package however now it looks like it's been subclassed / modified (I'm not great with Python) to support custom permissions and I can't really see whether it should still work anywhere (I have tried and can't seem to get it working).

Worth mentioning we have LDAP configured and working against our Primary domain without any issues.

**Comment by TGH-TH** [comment]
I find myself needing a similar config - any lessons learned before I roll my sleeves up?

**Comment by IdRatherStand** [comment]
Yes - I did manage to implement this thanks to the following Gist from @decoupca :
https://gist.github.com/decoupca/d4ceef8d1bfa1fed04b07eb87558f9a3

**Reply by TGH-TH** [reply]
Excellent - Thanks very much... much appreciated!
